datajunction-server/tests/api/nodes_test.py

```diff
@@ -555,6 +555,65 @@ def test_deactivating_linked_dimension(
             },
         )
         assert response.ok
+
+        # Create a metric on the source node w/ bound dimensions
+        response = client.post(
+            "/nodes/metric/",
+            json={
+                "description": "Total number of user messages by id",
+                "query": "SELECT COUNT(DISTINCT id) FROM default.messages",
+                "mode": "published",
+                "name": "default.num_messages_id",
+                "bound_dimensions": ["default.messages.id"],
+            },
+        )
+        assert response.ok
+
+        # Create a metric w/ bound dimensions that to not exist
+        with pytest.raises(Exception) as exc:
+            response = client.post(
+                "/nodes/metric/",
+                json={
+                    "description": "Total number of user messages by id",
+                    "query": "SELECT COUNT(DISTINCT id) FROM default.messages",
+                    "mode": "published",
+                    "name": "default.num_messages_id",
+                    "bound_dimensions": ["default.nothin.id"],
+                },
+            )
+            assert "bound dimensions that are not on parent nodes" in str(exc)
+
+        # Create a metric on the source node w/ an invalid bound dimension
+        response = client.post(
+            "/nodes/metric/",
+            json={
+                "description": "Total number of user messages by id",
+                "query": "SELECT COUNT(DISTINCT id) FROM default.messages",
+                "mode": "published",
+                "name": "default.num_messages_id_invalid_dimension",
+                "bound_dimensions": ["default.messages.foo"],
+            },
+        )
+        assert response.status_code == 400
+        assert response.json() == {
+            "message": (
+                "Node definition contains references to columns as "
+                "bound dimensions that are not on parent nodes."
+            ),
+            "errors": [
+                {
+                    "code": 206,
+                    "message": (
+                        "Node definition contains references to columns as "
+                        "bound dimensions that are not on parent nodes."
```

Comment:
Can we call these "required dimensions" rather than "bound" in the error messages? Mainly because we'll probably just surface this error directly to users, and I think the former is clearer naming-wise.